core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,643 @@
+                  annotationContent.setSource(((DAnnotation) cell).getSource() + "_CONTENT");
+                  annotationContent.getDetails().put(CONTENT, (String) obj);
+                  ((DRepresentation) diagram).getEAnnotations().add(annotationContent);
+                  list.add(annotationContent);
+                }
+              }
+            }
+          }
+        } catch (EvaluationException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+    return list;
+  }
+
+  public String getTitleBlockCellLabel(EObject diagram, EObject cell) {
+    if (cell instanceof DAnnotation) {
+      String name = ((DAnnotation) cell).getDetails().get(NAME);
+      if (name != null)
+        return name;
+    }
+    return "";
+  }
+
+  public String getCellLabel(EObject obj) {
+    if (obj instanceof DAnnotation) {
+      DAnnotation annotation = (DAnnotation) obj;
+      return annotation.getDetails().get(CONTENT);
+    }
+    return "";
+  }
+
+  public boolean isAnnotation(EObject obj) {
+    return obj instanceof DAnnotation;
+  }
+
+  public boolean isTitleBlockAllowed(EObject element, EObject containerView) {
+    if (containerView instanceof DRepresentation) {
+      return true;
+    }
+    return false;
+  }
+
+  public List<DAnnotation> getAvailableDiagramTitleBlocksToInsert(final EObject elementView) {
+    return getAvailableTitleBlocksToInsert(elementView, DIAGRAM_TITLE_BLOCK);
+  }
+
+  public List<DAnnotation> getAvailableElementTitleBlocksToInsert(final EObject elementView) {
+    return getAvailableTitleBlocksToInsert(elementView, ELEMENT_TITLE_BLOCK);
+  }
+
+  private List<DAnnotation> getAvailableTitleBlocksToInsert(final EObject elementView, String type) {
+    List<DAnnotation> result = new ArrayList<>();
+    EList<DAnnotation> eList = ((DDiagram) elementView).getEAnnotations();
+    for (DAnnotation elem : eList) {
+      if (elem.getSource().equals(type)) {
+        result.add(elem);
+      }
+    }
+    return result;
+  }
+
+  public List<DAnnotation> getInitialSelectionOfShowHideDiagramTitleBlocks(final EObject elementView) {
+    return getInitialSelectionOfShowHideTitleBlocks(elementView, DIAGRAM_TITLE_BLOCK);
+  }
+
+  public List<DAnnotation> getInitialSelectionOfShowHideElementTitleBlocks(final EObject elementView) {
+    return getInitialSelectionOfShowHideTitleBlocks(elementView, ELEMENT_TITLE_BLOCK);
+  }
+
+  private List<DAnnotation> getInitialSelectionOfShowHideTitleBlocks(final EObject elementView, String type) {
+    List<DAnnotation> result = new ArrayList<>(1);
+    DSemanticDiagram diagram = (DSemanticDiagram) CapellaServices.getService().getDiagramContainer(elementView);
+    if (elementView.equals(diagram)) {
+      EList<DAnnotation> diagramElements = diagram.getEAnnotations();
+      for (DAnnotation dDiagramElement : diagramElements) {
+        if (dDiagramElement.getSource().equals(type)) {
+          if (null != dDiagramElement.getDetails().get(VISIBILITY)) {

```

Comment:
For Show/Hide part, why do we need to use the attribute VISIBILITY to verify that a title block is visible or not? In other diagrams, we do not need this attribute to do Show/Hide.